library/core/src/slice/mod.rs

```diff
@@ -2253,11 +2294,11 @@ impl<T> [T] {
     pub fn rotate_left(&mut self, mid: usize) {
         assert!(mid <= self.len());
         let k = self.len() - mid;
+        let p = self.as_mut_ptr();
 
-        // SAFETY: `[mid - mid;mid+k]` corresponds to the entire
+        // SAFETY: `[mid; mid+k]` corresponds to the entire
         // `self` slice, thus is valid for reads and writes.
```

Comment:
I still don't quite understand this :/ 

`ptr_rotate` says, that the range `[mid-left, mid+right)` needs to be valid for reading and writing. The `mid` in that function is our `p.add(mid)`, `left` is `mid` and `right` is `k`. Maybe just say something like:

> The range `[p.add(mid) - mid, p.add(mid) + k)` is trivially valid for reading and writing, as required by `ptr_rotate`.

Reply:
Yup, your version is better. I understand the original version well but maybe it's because I read the code several times and investigated more. Thanks !